src/main/kotlin/uk/gov/dluhc/printapi/mapper/ElectoralRegistrationOfficeMapper.kt

```diff
@@ -0,0 +1,39 @@
+package uk.gov.dluhc.printapi.mapper
+
+import org.mapstruct.Mapper
+import org.mapstruct.Mapping
+import uk.gov.dluhc.printapi.database.entity.ElectoralRegistrationOffice
+import uk.gov.dluhc.printapi.dto.EroContactDetailsDto
+import uk.gov.dluhc.printapi.messaging.models.CertificateLanguage
+import uk.gov.dluhc.printapi.messaging.models.CertificateLanguage.CY
+import uk.gov.dluhc.printapi.messaging.models.CertificateLanguage.EN
+
+private const val ELECTORAL_REGISTRATION_OFFICER_EN = "Electoral Registration Officer"
+private const val ELECTORAL_REGISTRATION_OFFICER_CY = "Swyddog Cofrestru Etholiadol"
+
+@Mapper
+abstract class ElectoralRegistrationOfficeMapper {
+    @Mapping(target = "name", expression = "java( getName(language) )")
+    protected abstract fun toElectoralRegistrationOfficeFromNotNullContactDetails(
```

Comment:
Unfortunately, with the MapStruct implemantation, this method doesn't return null when the passed `eroContactDetails` is passed as null. Instead, it generates an instance of the `ElectoralRegistrationOffice`, with the field values set to null. That's why I added the `toElectoralRegistrationOffice`, which calls this method if `eroContactDetails` is not null and otherwise returns null.

Happy to implement a better solution, if there is any.

Here is the code generated by MapStruct:

<pre>
 @Override
    protected ElectoralRegistrationOffice toElectoralRegistrationOfficeFromNotNullContactDetails(EroContactDetailsDto eroContactDetails, CertificateLanguage language) {
        if ( eroContactDetails == null && language == null ) {
            return null;
        }

        ElectoralRegistrationOffice electoralRegistrationOffice = new ElectoralRegistrationOffice();

        if ( eroContactDetails != null ) {
            electoralRegistrationOffice.setPhoneNumber( eroContactDetails.getPhoneNumber() );
            electoralRegistrationOffice.setEmailAddress( eroContactDetails.getEmailAddress() );
            electoralRegistrationOffice.setWebsite( eroContactDetails.getWebsite() );
            electoralRegistrationOffice.setAddress( addressDtoToAddress( eroContactDetails.getAddress() ) );
        }
        electoralRegistrationOffice.setName( getName(language) );

        return electoralRegistrationOffice;
    }

</pre>